### ISSUE TITLE
Emulator crashing when trying to save and load state

### ISSUE THREAD

**Issue by Metallix01** [mgba-emu/mgba#3052]
Every time I save or load a file in this emulator, the whole thing crashes, not my pc fortunately, but the emulator closes suddenly, ironically enough, the file does get saved, is just that the game closes after doing that.

[Report file.zip](https://github.com/mgba-emu/mgba/files/13186081/Report.file.zip)

**Comment by anewuser** [comment]
Hi. Try selecting `OpenGL (force version 1.x)` as your display driver:

![screenshot](https://github.com/mgba-emu/mgba/assets/6733770/ded8fff9-8dec-470b-9892-78db72148f24)

**Comment by endrift** [comment]
That's bad advice. If the emulator works before loading or saving, this won't help.

**Comment by anewuser** [comment]
@endrift  My advice was based on an actual test on similar hardware. Please refrain from leaving negative comments that do not add anything to the discussion.

**Comment by endrift** [comment]
Are you aware you're chastising the lead dev?

**Comment by endrift** [comment]
Can you see if a development build (https://mgba.io/downloads.html#development-downloads) works properly?

**Comment by anewuser** [comment]
I have a laptop with a CPU from the same exact family as @Metallix01's. It uses the first iteration of Intel HD Graphics too. mGBA with default settings — including the latest development build — has always crashed while saving/loading states since I first tested it on this laptop years ago. 

There was an inconclusive bug report about this a while ago, but the user who filed it unfortunately closed it before any progress had been made. I didn't comment on it or file a new issue because I didn't want to bother the developers with something that was unlikely to affect anybody else, so I was surprised to see this new bug report.

The only workaround that I've found is going to the mGBA settings and changing the display driver to `OpenGL (force version 1.x)`. This disables the "Shaders" feature, but makes it possible for me to save and load states on this laptop normally.

**Comment by endrift** [comment]
I tried reproducing last night by forcing Mesa to use similar vintage OpenGL versions, but was unable to get it to crash. This may be a driver bug, but I can't confirm that without a crash dump. Can you try to get one out of the Windows Event Viewer? I have a custom XML filter to aid in this, which I've attached.

[crashes-view.zip](https://github.com/mgba-emu/mgba/files/13257782/crashes-view.zip)

**Comment by anewuser** [comment]
```
Faulting application name: mGBA.exe, version: 0.11.0.0, time stamp: 0x00000000
Faulting module name: ig4icd64.dll, version: 8.15.10.2900, time stamp: 0x50b4043d
Exception code: 0xc0000005
Fault offset: 0x00000000001b8111
Faulting process id: 0x1e64
Faulting application start time: 0x01da0fe5d9b9e200
Faulting module path: C:\WINDOWS\SYSTEM32\ig4icd64.dll

```

v8.15.10.2900 is the driver installed by Windows 10 automatically. Using the unsupported Windows 7 [v8.15.10.2993 driver](https://www.intel.com/content/www/us/en/download/15549/intel-graphics-media-accelerator-driver-windows-7-64-windows-vista-64-exe.html)  makes no difference.

It seems that running programs that use OpenGL on Windows 10 with these old CPUs has caused people all sorts of issues: [save-legacy-intel-graphics](https://github.com/pal1000/save-legacy-intel-graphics#readme).

**Comment by endrift** [comment]
Yep, that's a graphics driver bug alright...that's gonna be a difficult one to work around.
